### ISSUE TITLE
Training Siamese network is x300 slower than regular

### ISSUE THREAD

**Issue by oak-tree** [fchollet/keras#5657]
Hey,
I'm using both `tensorflow` and `theano`. Although for `big modules` `tensorflow` is much better - building time wise.

I'm using the lastest `keras` version (`master` branch) and `tensorflow` version 1

I have a dataset around 7000 images. `Theano` run a single apoch in around 70 seconds.`tensorwflow` runs this apoch around 800 seconds. 

What can cause such a huge difference comparing `tensorflow`?

Thanks

**Comment by PPACI** [comment]
I don't really know why tensorflow is slower than theano but there is one thing you might try.
since you're using TF 1.0, you can take advantage of XLA JIT compilation which should help in case like yours.
https://www.tensorflow.org/versions/master/experimental/xla/jit

All you need to do if you're planning on a basic usage of XLA is to add the following to your code before creating the model : 
```python
import keras.backend.tensorflow_backend as bck
config = bck.tf.ConfigProto()
config.graph_options.optimizer_options.global_jit_level = bck.tf.OptimizerOptions.ON_1
bck.set_session(bck.tf.Session(config=config))
model = Sequential() ...
```
Tell us if it helped

**Comment by oak-tree** [comment]
very interesting stuff, ill check it out. Do you think that single gpu
macine can also benefit from it?

**Comment by PPACI** [comment]
It can, depending on your network architecture. Complex network tend to benefit more from XLA than a simple sequential.

**Comment by oak-tree** [comment]
Hey @PPACI 

Two questions :

1. I have tried the lines. It still does not reach `theano` speed. But I'm wonder how can I verify it ineeded use the `JIT` engine ?

1. by the way if this all it takes to set up jit maybe it worth to create include file: ["usetensorjit" for example](https://gist.github.com/oak-tree/7f1d57a52b1e439a79f73c072adcf2f6).  Then all it takes to make keras use it is jut import it before building the model it self. What do you think?
